plugins/outputs/cloudwatch/cloudwatch_test.go

```diff
@@ -59,6 +59,131 @@ func TestBuildDimensions(t *testing.T) {
 	}
 }
 
+// Test that we retain as many tags as possible if the number exceeds the max number permitted
+func TestBuildDimensionsExceedMaxNumber(t *testing.T) {
+	assert := assert.New(t)
+
+	testPoint := testutil.TestMetric(1)
+
+	for i := len(testPoint.Tags()); i <= MaxDimensions; i++ {
+		testPoint.AddTag(fmt.Sprint(i), "Value"+fmt.Sprint(i))
+	}
+
+	dimensions := BuildDimensions(testPoint.Tags(), nil)
+
+	assert.Equal(MaxDimensions, len(dimensions), "Number of dimensions should be less than MaxDimensions")
+	assert.Equal(len(testPoint.Tags())-1, len(dimensions), "Should keep as many dimensions as possible")
+
+	for _, dimension := range dimensions {
+		tagValue := testPoint.Tags()[*dimension.Name]
+		assert.Equal(tagValue, *dimension.Value, "Value should be equal")
+	}
+}
+
+// Test that global dimensions are added correctly
+func TestGlobalDimensions(t *testing.T) {
+	assert := assert.New(t)
+
+	testPoint := testutil.TestMetric(1)
+	globalDimensions := map[string]string{
+		"Environment": "test",
+		"Deployment":  "green",
+	}
+	dimensions := BuildDimensions(testPoint.Tags(), globalDimensions)
+
+	tagKeys := make([]string, len(testPoint.Tags()))
+	i := 0
+	for k := range testPoint.Tags() {
+		tagKeys[i] = k
+		i += 1
+	}
+
+	sort.Strings(tagKeys)
+
+	assert.Equal(len(testPoint.Tags())+len(globalDimensions), len(dimensions), "Number of dimensions should be equal to number of tags plus the number of global dimensions")
+
+	for key, value := range globalDimensions {
+		var matchingDimension *cloudwatch.Dimension
+
+		for _, dimension := range dimensions {
+			if *dimension.Name == key {
+				matchingDimension = dimension
+			}
+		}
+
+		assert.NotNil(matchingDimension, "Global dimension should exist")
+		assert.Equal(value, *matchingDimension.Value, "Global dimension value should be equal")
+	}
+
+	for i, key := range tagKeys {
+		assert.Equal(key, *dimensions[i+len(globalDimensions)].Name, "Key should be equal")
+		assert.Equal(testPoint.Tags()[key], *dimensions[i+len(globalDimensions)].Value, "Value should be equal")
+	}
+}
+
+// Test that global dimensions are added correctly
+func TestGlobalDimensionsExceedMaxNumber(t *testing.T) {
```

Comment:
IMO, it would be better to have a table driven test to [test different behaviors of global dimensions here](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) . Your thoughs?